src/cosalib/cli.py

```diff
@@ -0,0 +1,119 @@
+        self.add_env_arg(
+            'COSA_LOG_LEVEL', 'info', '--log-level',
+            choices=LOG_LEVELS, help='Set the log level')
+
+    def add_env_arg(self, env_var, default, *args, **kwargs):
+        """
+        Adds an argument that can also pull it's value from an
+        environment variable.
+
+        :param env_var: The environment variable to look at
+        :type env_var: str
+        :env default: The default for the argument
+        :type default: any
+        :param args: Non keyword arguments to pass to add_argument
+        :type args: list
+        :param kwargs: Keyword arguments to pass to add_argument
+        :type kwargs: dict
+        """
+        if kwargs.get('help') is None:
+            kwargs['help'] = ''
+        kwargs['help'] = kwargs['help'] + ' (Env: {})'.format(env_var)
+        self.parser.add_argument(
+            *args, default=os.environ.get(env_var, default), **kwargs)
+
+    def parse_args(self):
+        """
+        Parses the arguments passed in, verifies inputs, sets the logger,
+        and returns the arguments.
+
+        :returns: The parsed arguments
+        :rtype: argparse.Namepsace
+        """
+        args = self.parser.parse_args()
+        self._set_logger(args.log_level)
+        return args
+
+    def _set_logger(self, level):
+        """
+        Set the log level
+
+        :param level: set the log level
+        :type level: str
+        """
+        sl = log.DEBUG
+        if level == "error":
+            sl = log.ERROR
+        elif level == "warn":
+            sl = log.WARN
+        elif level == "info":
+            sl = log.INFO
+
+        log.basicConfig(
+            format='[%(asctime)s  %(levelname)s]: %(message)s',
+            level=sl)
+
+    def add_argument(self, *args, **kwargs):
+        """
+        Forward to self.parser.add_argument.
+
+        :param args: All non-keyword arguments
+        :type args: list
+        :param kwargs: All keyword arguments
+        :type kwargs: dict
+        :returns: The argument action
+        :rtype: argparse.Action
+        """
+        return self.parser.add_argument(*args, **kwargs)
+
+    # Forwards for common parser usage
+    print_help = lambda s: s.parser.print_help()  # noqa: E731
+    print_usage = lambda s: s.parser.print_usage()  # noqa: E731
+    get_default = lambda s, dest: s.get_default(dest)  # noqa: E731
+    error = lambda s, msg: s.parser.error(msg)  # noqa: E731
+    exit = lambda s, code=0: s.parser.exit(code)  # noqa: E731
+
+
+class BuildCli(Cli):
+    """
+    Cli class that adds in reusable build specific arguments.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializes the BuildCli instance.
+
+        :param kwargs: All keyword arguments which will pass to ArgumentParser
+        :type kwargs: dict
+        """
+        Cli.__init__(self, **kwargs)
+        # Set common arguments
+        self.parser.add_argument(
```

Comment:
Why not use the `add_argument` helper function over calling it on the `parser` variable directly?

Reply:
No good reason! Updating :laughing: